service/service.go

```diff
@@ -270,6 +276,23 @@ func (app *Application) setupExtensions(ctx context.Context) error {
 	return app.builtExtensions.StartAll(ctx, app)
 }
 
+func (app *Application) createExportersMap() {
+	app.exportersMap = make(map[configmodels.DataType]map[configmodels.Exporter]component.Exporter)
+	app.exportersMap[configmodels.TracesDataType] = make(map[configmodels.Exporter]component.Exporter)
+	app.exportersMap[configmodels.MetricsDataType] = make(map[configmodels.Exporter]component.Exporter)
+
+	for cfg, exp := range app.builtExporters {
+		te := exp.GetTraceExporter()
+		if te != nil {
+			app.exportersMap[configmodels.TracesDataType][cfg] = te
+		}
+		me := exp.GetMetricExporter()
+		if me != nil {
+			app.exportersMap[configmodels.MetricsDataType][cfg] = me
+		}
+	}
+}
```

Comment:
This can be in the builder file and just return `map[configmodels.DataType]map[configmodels.Exporter]component.Exporter`

Reply:
Done.